extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -0,0 +1,490 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+import axios, { AxiosResponse } from 'axios';
+import * as qs from 'qs';
+import * as url from 'url';
+
+import {
+	AzureAccountProviderMetadata,
+	Tenant,
+	AzureAccount,
+	Resource,
+	AzureAuthType,
+	Subscription
+} from '../interfaces';
+
+import { SimpleTokenCache } from '../simpleTokenCache';
+import { MemoryDatabase } from '../utils/memoryDatabase';
+const localize = nls.loadMessageBundle();
+
+export interface AccountKey {
```

Comment:
Not sure if those comments are useful unless they provide some information about the fields

Reply:
True haha

Reply:
Similar cases in the interfaces below :)

Reply:
The other ones seem pretty self explanatory :D